139_Client_Server/python_asyncio_server.py

```diff
@@ -0,0 +1,72 @@
+import socket 
+import asyncio 
+
+# Thread for each client
+async def serve(client_connect, client_address, client_id, clients):
+    """Serve a client."""
+    loop = asyncio.get_event_loop()
+    try:
+        print(f"Serve {client_id}")
+        while True:
+            await loop.sock_sendall(client_connect,b'Write something!')
+            data = (await loop.sock_recv(client_connect, 80)).decode('utf8')
+            print(f"Got from {client_id} : {data}")
+            if data.lower() == 'quit':
+                print(f"{client_id} has left!")
+                break
+            else:
+                await loop.sock_sendall(client_connect,
+                                        b'You sent %s' % data.encode('utf8'))
+    finally:
+        #Close Connection
+        client_connect.close()
+        clients.remove(client_id)
+        print_clients(clients)
+
+#Function to print current clients
+def print_clients(clients):
+    """Print client list and number of connections."""
+    print("Number of current Connections:", len(clients))
+    print("List of clients:", clients)
+
+#Function to establish a TCP host
+async def server():
+        """Accept many clients forever."""
+        loop = asyncio.get_event_loop()
+        clients = []
+        clientLoops = []
```

Comment:
This variable seems unused (and should have been named `client_loops`.
Also the whole block is indented 8 spaces instead of 4.